dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityPreviewServlet.java

```diff
@@ -0,0 +1,359 @@
+            request.getRequestDispatcher("/portal/no_license.jsp")
+                .forward(request, response);
+            return;
+        }
+
+        if (DbConnectionFactory.isOracle() && LicenseUtil.getLevel() < LicenseLevel.PRIME.level) {
+            request.getRequestDispatcher("/portal/no_license.jsp")
+                .forward(request, response);
+            return;
+        }
+        if (!LicenseUtil.isASAllowed()) {
+            request.getRequestDispatcher("/portal/no_license.jsp")
+                .forward(request, response);
+            return;
+
+        }
+
+        try {
+
+
+            if (uri == null) {
+                response.sendError(500, "VelocityServlet called without running through the CMS Filter");
+                Logger.error(this.getClass(),
+                        "You cannot call the VelocityServlet without passing the requested url via a  requestAttribute called  "
+                                + Constants.CMS_FILTER_URI_OVERRIDE);
+                return;
+            }
+
+
+
+            // we will always need a visitor in admin mode
+            if (mode.isAdmin) {
+                visitorAPI.getVisitor(request, true);
+            }
+
+
+            switch (mode) {
+                case PREVIEW_MODE:
+                    Logger.debug(VelocityPreviewServlet.class, "VELOCITY SERVLET I'M ON PREVIEW MODE!!!");
+                    doPreviewMode(request, response);
+                    break;
+                case EDIT_MODE:
+                    Logger.debug(VelocityPreviewServlet.class, "VELOCITY SERVLET I'M ON EDIT MODE!!!");
+                    doEditMode(request, response);
+                    break;
+                default:
+                    response.sendError(HttpServletResponse.SC_UNAUTHORIZED);
+                    return;
+            }
+
+
+
+        } catch (ResourceNotFoundException rnfe) {
+            Logger.error(this, "ResourceNotFoundException" + rnfe.toString(), rnfe);
+            response.sendError(404);
+            return;
+        } catch (ParseErrorException pee) {
+            Logger.error(this, "Template Parse Exception : " + pee.toString(), pee);
+            response.sendError(500, "Template Parse Exception");
+        } catch (MethodInvocationException mie) {
+            Logger.error(this, "MethodInvocationException" + mie.toString(), mie);
+            response.sendError(500, "MethodInvocationException Error on template");
+        } catch (Exception e) {
+            Logger.error(this, "Exception" + e.toString(), e);
+            response.sendError(500, "Exception Error on template");
+
+        } finally {
+            DbConnectionFactory.closeSilently();
+        }
+
+    }
+
+    public void init(ServletConfig config) throws ServletException {
+
+
+
+    }
+
+    public void doPreviewMode(HttpServletRequest request, HttpServletResponse response) throws Exception {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)